rest_framework_extensions/fields.py

```diff
@@ -25,4 +26,38 @@ class Meta:
 
     def __init__(self, *args, **kwargs):
         kwargs.setdefault('source', '*')
-        super(ResourceUriField, self).__init__(*args, **kwargs)
\ No newline at end of file
+        super(ResourceUriField, self).__init__(*args, **kwargs)
+
+
+class MappingField(Field):
+    default_error_messages = {
+        'key_not_found': '"{value}" not found in "mapping" keys',
+        'value_not_found': '"{value}" not found in "mapping" values'
+    }
+
+    def __init__(self, mapping, reverse_mapping=None, **kwargs):
+        super(MappingField, self).__init__(**kwargs)
+
+        def check_dict(d):
+            assert isinstance(d, dict), '"mapping" should be a dictionary'
+            for k, v in d.items():
+                assert isinstance(k, (str, int, bool)) and isinstance(v, (str, int, bool)), '"mapping" can contain only simple types'
+
+        check_dict(mapping)
+
+        self.mapping = mapping
+        if reverse_mapping is None:
+            self.reverse_mapping = {v: k for k, v in mapping.iteritems()}
+        else:
+            check_dict(reverse_mapping)
+            self.reverse_mapping = reverse_mapping
+
+    def to_representation(self, value):
+        if value in self.mapping:
+            return self.mapping[value]
+        self.fail('key_not_found', value=value)
+
+    def to_internal_value(self, data):
+        if data in self.reverse_mapping:
+            return self.reverse_mapping[data]
+        self.fail('value_not_found', value=data)
```

Comment:
no new line at the end (this error could be checked automaticaly by using flake8)